docs/src/main/sphinx/functions/comparison.md

```diff
@@ -301,3 +301,47 @@ WHERE regionkey IN (
     FROM region
 );
 ```
+
+## Examples
+
+The following example queries showcase aspects of using comparison functions and
+operators related to implied ordering of values, implicit casting, and different
+types.
+
+Ordering:
+
+```sql
+SELECT 'M' BETWEEN 'A' AND 'Z'; -- true
+SELECT 'A' < 'B'; -- true
+SELECT 'A' < 'a'; -- true
+SELECT TRUE > FALSE; -- true
+SELECT 'M' BETWEEN 'A' AND 'Z'; -- true
+SELECT 'm' BETWEEN 'A' AND 'Z'; -- false
+```
+
+The following queries show a subtle difference between `char` and `varchar`
+types. The length parameter for `varchar` is an optional maximum length
+parameter and comparison happens directly with the data:
```

Comment:
What does "comparison happens directly with the data" mean?

Reply:
No casting or whatever.. maybe

comparison happens without automatic conversions: